extensions/service_bindings_manager.go

```diff
@@ -0,0 +1,76 @@
+package extensions
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// ServiceBinding is a representation of a service binding that has
+// been provided into the build container.
+type ServiceBinding struct {
+	// Name is the name of the directory that binding was contained within.
+	Name string
+
+	// Type is the content of the type file contained within the binding
+	// directory.
+	Type string
+
+	// Provider is the content of the provider file contained within the binding
+	// directory.
+	Provider string
+
+	// Path is the filepath of the binding directory.
+	Path string
+
+	// Secrets is a mapping of file name to contents for all other files
+	// contained in the bindings directory.
+	Secrets map[string][]byte
+}
+
+// ListServiceBindings returns all service bindings found in the given root
+// directory. These bindings must conform the the Kubernetes Service Binding
+// specification as documented in https://github.com/k8s-service-bindings/spec.
+func ListServiceBindings(root string) ([]ServiceBinding, error) {
+	directories, err := filepath.Glob(filepath.Join(root, "*"))
```

Comment:
`ioutil.ReadDir` might be more clear than a custom glob here.

Also, it might be prudent to filter out non-directories here

Reply:
@menehune23 `ioutil` is technically deprecated, even though it hasn't been removed from Go. Because of that, we tend to stay away from it when we can